File: frontend/src/services/authService.ts
import { api } from '@/lib/axios';

export interface LoginRequest {
  email: string;
  password: string;
}

export interface LoginResponse {
  access_token: string;
  refresh_token: string;
  token_type: string;
  user: any; // Build proper user type
}

export const authService = {
  login: async (credentials: LoginRequest): Promise<LoginResponse> => {
    // FastAPI OAuth2 requires form data
    const params = new URLSearchParams();
    params.append('username', credentials.email);
    params.append('password', credentials.password);

    const response = await api.post<LoginResponse>('/login/access-token', params, {
      headers: {
        'Content-Type': 'application/x-www-form-urlencoded',
      },
    });

    const loginData = response.data;

    // Fetch user details separately if not included in login response
    const user = await authService.getCurrentUser();

    return {
      ...loginData,
      user,
    };
  },

  logout: async () => {
    await api.post('/auth/logout');
  },

  getCurrentUser: async () => {
    const response = await api.get('/users/me');
    return response.data;
  },
};

File: mobile/driver_app/lib/features/auth/data/auth_service.dart
import 'package:dio/dio.dart';

class LoginResponse {
  final String accessToken;
  final String tokenType;

  LoginResponse({required this.accessToken, required this.tokenType});

  factory LoginResponse.fromJson(Map<String, dynamic> json) {
    return LoginResponse(
      accessToken: json['access_token'],
      tokenType: json['token_type'],
    );
  }
}

class AuthService {
  final Dio _dio;

  AuthService(this._dio);

  Future<LoginResponse> login(String username, String password) async {
    // The backend expects application/x-www-form-urlencoded
    final data = {'username': username, 'password': password};

    final response = await _dio.post(
      '/login/access-token', // Relative to AppConstants.baseUrl
      data: data,
      options: Options(contentType: Headers.formUrlEncodedContentType),
    );

    return LoginResponse.fromJson(response.data);
  }

  /// Update FCM token on the backend
  Future<void> updateFcmToken(String token) async {
    await _dio.post(
      '/api/v1/drivers/me/fcm-token',
      data: {'fcm_token': token},
    );
  }
}
